### PR TITLE
[new release] jose (0.1.0)

### PR DESCRIPTION
**Pull request by ulrikstrid**
JOSE implementation for OCaml and ReasonML

- Project page: <a href="https://ulrikstrid.github.io/reason-jose">https://ulrikstrid.github.io/reason-jose</a>
- Documentation: <a href="https://ulrikstrid.github.io/reason-jose">https://ulrikstrid.github.io/reason-jose</a>

##### CHANGES:

- Initial release

### CODE REVIEW

**Comment by ulrikstrid** [comment]
I realize I have a bunch of issues here. `rely` isn't on opam and I need a unreleased version of `nocrypto`.
Should I close this PR or change `rely` to something else and hope that `nocrypto` creates a new release and update this?

File: packages/jose/jose.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+maintainer: ["ulrik.strid@outlook.com"]
+authors: ["Ulrik Strid"]
+license: "MIT"
+homepage: "https://ulrikstrid.github.io/reason-jose"
+doc: "https://ulrikstrid.github.io/reason-jose"
+bug-reports: "https://github.com/ulrikstrid/reason-jose/issues"
+depends: [
+  "ocaml" {>= "4.06.0"}
+  "base64"
+  "dune" {>= "1.11" & < "2.0"}
```
Comment (kit-ty-kate) [inline packages/jose/jose.0.1.0/opam]: I'm intrigued, what kind of problem did you encountered to not be compatible with dune 2.0 if you can elaborate on that?

File: packages/jose/jose.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "-j"
+    jobs
+    "@install"
+    "@doc" {with-doc}
+  ]
+  ["dune" "exec" "test/test_exe/RunTests.exe"] {with-test}
+]
+pin-depends: [
+  [ "nocrypto.dev" "git+https://github.com/mirleft/ocaml-nocrypto.git#dune" ]
+]
```
Comment (kit-ty-kate) [inline packages/jose/jose.0.1.0/opam]: pin-dependencies are forbidden in opam-repository.

Comment (kit-ty-kate) [reply packages/jose/jose.0.1.0/opam]: This field should be removed.

File: packages/jose/jose.0.1.0/opam [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@doc" {with-doc}
+  ]
+  ["dune" "exec" "test/test_exe/RunTests.exe"] {with-test}
```
Comment (kit-ty-kate) [inline packages/jose/jose.0.1.0/opam]:
```suggestion
  ["dune" "exec" "-p" name "test/test_exe/RunTests.exe"] {with-test}
```
In releases, dune should always be called with `-p name` or `--profile=release` as the default (dev) profile might turn new warnings into errors in the future.

**Comment by kit-ty-kate** [comment]
> Should I close this PR or change `rely` to something else and hope that `nocrypto` creates a new release and update this?

We can only merge packages that are installable and released, I'm not sure what `rely` is but at least `nocrypto` needs to be released first as said above (sorry I only saw your comment afterwards)

File: packages/jose/jose.0.1.0/opam (hunk 1 above)
Comment (ulrikstrid) [reply packages/jose/jose.0.1.0/opam]: Nocrypto's dune branch wasn't up-to-date and used `dune.configuration` which doesn't exist in 2.0. This is a bandaid for that basically, but this should be changed when nocrypto has a proper release.

**Comment by ulrikstrid** [comment]
superseded by #16006

**Pull request closed without merging**